pkg/pxc/app/binlogcollector/binlog_collector_test.go

```diff
@@ -0,0 +1,170 @@
+package binlogcollector
+
+import (
+	"reflect"
+	"testing"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetDeploymentLabelsAndMatchLabels(t *testing.T) {
+	createCR := func() *api.PerconaXtraDBCluster {
+		return &api.PerconaXtraDBCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "default",
+			},
+			Spec: api.PerconaXtraDBClusterSpec{
+				CRVersion: "1.15.0",
```

Comment:
For fetching the latest version, I think the right approach to ensure that the tests will remain updated and not forget to change the hardcoded version, to use the `version.Version()' functionality we recently introduced.